client/starwhale/core/dataset/store.py

```diff
@@ -638,7 +638,7 @@ def read(self, size: int) -> bytes:
             return b""
         elif size < 0:
             # urllib3 resp read does not support to read -1
-            return self.resp.raw.read()  # type: ignore
+            return self.resp.content  # type: ignore
```

Comment:
`self.resp.content = self.resp.raw.read()` and use the uniform code schema for the different sizes (size <0 and size >= 0).

Reply:
sometimes `self.resp.content != self.resp.raw.read()`
```python
>>>import requests
>>> resp=requests.get("https://starwhale-examples.oss-cn-beijing.aliyuncs.com/dataset/kitti/lane/data_road/training/calib/um_000000.txt",stream=True)
>>> b=resp.raw.read()
>>> len(b)
502
>>> resp=requests.get("https://starwhale-examples.oss-cn-beijing.aliyuncs.com/dataset/kitti/lane/data_road/training/calib/um_000000.txt",stream=True)
>>> b=resp.content
>>> len(b)
1860
>>> 

```

Reply:
the streaming mode maybe led the unequal value.

Reply:
> the streaming mode maybe led the unequal value.

Without streaming mode we can't access `resp.raw.read()` like this. See the doc: https://requests.readthedocs.io/en/latest/user/quickstart/#raw-response-content

> In the rare case that you’d like to get the raw socket response from the server, you can access r.raw. If you want to do this, make sure you set stream=True in your initial request. Once you do, you can do this:
> r = requests.get('https://api.github.com/events', stream=True)
> r.raw
> <urllib3.response.HTTPResponse object at 0x101194810>
> r.raw.read(10)
> '\x1f\x8b\x08\x00\x00\x00\x00\x00\x00\x03'